README.md

```diff
@@ -86,7 +86,7 @@ VIN: Vietnames
 ### Using CLI
 
 ```bash
-# `--use-cuda` is optional, if you have a compatible NVIDIA GPU, you can use it. (To GPU accelerate offline translation set the USE_CUDA_FOR_OFFLINE_TRANSLATION env var to true)
```

Comment:
This was originally to limit overall GPU memory usage for loaded models. Given GPU usage sits ~ 7 - 8 GB without this model loaded which is the limit of most high end desktop GPU's.

I would prefer an option to load this model into RAM while still using CUDA for others if at all possible as it blocks the vast majority of mid range graphics cards (even with the small model). Though it does lazy load so might not be a problem. 🤔

Reply:
how about adding a flag instead of using an environment variable then:
```py
parser.add_argument('--use-cuda', action='store_true', help='turn on/off cuda (excluding offline translator)')
parser.add_argument('--use-cuda-full', action='store_true', help='turn on/off cuda')
```

Reply:
Possibly, I think part of the problem was that the arguments never reach the child process forked for the web process. You might need to find a way to pass that down as environment variables to reach the child process scope (Part of why it originally was an env var :laughing: ) but I don't think CLI flags ever do. (See https://github.com/zyddnys/manga-image-translator/blob/5570b181374936a321547aa6722c6cc121303edc/translate_demo.py#L307)

You will need to find a way got thoes arguments into the context of the child process so that they are not ignored when running in web mode :eyes:  @thatDudo . 

Thanks for having a look!

Reply:
Is that really a problem? The child process is not the one who run the image translation anyway, thats the job of `translate_demo.py` so the user would be able to specify the cuda usage on startup.
Also, before me adding the flag i checked for any usages of `USE_CUDA_FOR_OFFLINE_TRANSLATION` and didnt find anything, so it should be ok. Regardless I think its more practical to keep the arguments at one place and to make `--use_cuda` actually make the program use cuda everywhere like you'd think it would.

Reply:
I am pretty sure the translation task runs on the web process. Dumping the current argv of the running process (`sys.argv`) does indicate the executing process when using the process translation in web mode:
```
translator youdao
target_language ENG
['web_main.py', '6f7d6c7d59854aac632ba5adc100ed99', '0.0.0.0', '5003']
```

It does not use an external process when running in batch/demo mode.  

Environment variables work because they are passed directly to the child processes scope by default when using the popen constructor  
```
default behavior of inheriting the current process’ environment.
```
https://docs.python.org/3.8/library/subprocess.html#popen-constructor.

As you raise it is certainly better to pass it as a CLI flag to the process at startup 😆  

But it will need to reach the child processes' context in order for the argument to be picked up.

Reply:
Oh yeah youre right wow, thanks for pointing out. Interestingly it is running the translation in parallel to the inpainting, which is fine if youre not running the offline translator but not good otherwise... I will probably have to move the machine translation part back to `translate_demo.py` then. Youre so helpful thank you :smile:.

Reply:
It should be ready now

Reply:
I decided to move the offline translator to the main process because:
- There should not be multiple models running at the same time
- So that the cuda flag does not have to be passed to the web process (too dirty a solution for me)

I understand that using an environment variable makes it simpler to pass the cuda usage level to the offline translator but I do also think that its intuitive to the user that thinks that `--use-cuda` will make the program use cuda overall without further intervention.
Its best to keep things at one place or things will get messy quickly and no one is going to know whats going on in this project.

Reply:
Looks great from my perspective <3